main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+    "flag"
+    "fmt"
+    "./schemas"
+    "github.com/companieshouse/chs.go/avro"
+    "github.com/companieshouse/chs.go/kafka/producer"
+    "github.com/companieshouse/chs.go/avro/schema"
+    "os"
+    "os/signal"
+    "github.com/Shopify/sarama"
+    "strings"
+    "strconv"
+)
+
+// Assigns all flags to variables
+var (
+    brokerPtr = flag.String("broker", "", "Broker")
+    topicPtr = flag.String("topic", "", "Topic name")
+    schemaPtr = flag.String("schema", "", "Schema")
+    schemaRegistryPtr = flag.String("schema-registry", "", "Schema Registry")
+    partitionPtr = flag.Int64("partition", 0, "Partition")
+    zookeeperPtr = flag.String("zookeeper", "", "Zookeeper")
+    offsetPtr = flag.String("offset", "-1", "Offset number")
+    jsonOutPtr = flag.Int("json-out", -1, "Does deserialized JSON get printed to terminal?")
+)
+
+func main () {
+    flag.Parse()
```

Comment:
Consider using `flag.PrintDefaults()` at this point to show a usage message describing the flags (and their default values), after testing whether the minimum mandatory flags have been provided or not.

Reply:
Is it worth putting it as a condition? For example, if the flag `-help` has been passed, to then execute the `PrintDefaults()`? This way, when the user has used the tool correctly, the `PrintDefaults()` isn't printed out as well therefore minimising clutter in the terminal.